website/content/docs/k8s/installation/vault/data-integration/webhook-certs.mdx

```diff
@@ -0,0 +1,261 @@
+
+### Bootstrapping the PKI Engine for Controller and Connect Inject Webhooks
+
+First, we need to bootstrap the Vault cluster by enabling and configuring the PKI Secrets Engine to be able to serve TLS certificates for the controller and connect inject webhooks.
+
+* Mount the PKI Secrets Engine for each:
+
+  ```shell-session
+  $ vault secrets enable -path=controller pki
+  ```
+
+  ```shell-session
+  $ vault secrets enable -path=connect-inject pki
+  ```
+
+* Tune the engine mounts to enable longer TTL:
+
+  ```shell-session
+  $ vault secrets tune -max-lease-ttl=87600h controller
+  ```
+
+  ```shell-session
+  $ vault secrets tune -max-lease-ttl=87600h connect-inject
+  ```
+
+* Generate the root CA for each:
+
+  ```shell-session
+  $ vault write -field=certificate controller/root/generate/internal \
+          common_name="<RELEASE NAME>-controller-webhook" \
+          ttl=87600h
+  ```
+
+  ```shell-session
+  $ vault write -field=certificate connect-inject/root/generate/internal \
+          common_name="<RELEASE NAME>-connect-injector" \
+          ttl=87600h
+  ```
+
+
+## Overview
+In a Consul Helm chart configuration that does not use Vault, webhook-cert-manager normally fulfills the role of ensuring that a valid certificate is updated to the mutatingwebhook configuration of either controller or connect inject to ensure that Kubernetes can communicate with each of these services. 
+
+When Vault is configured as the controller and connect inject Webhook Certificate Provider on Kubernetes: 
+  - webhook-cert-manager is no longer deployed to the cluster.
+  - controller and connect inject each get their webhook certificates from its own Vault PKI mount via the injected Vault Agent.
+  - controller and connect inject each need to be configured with its own Vault Role that has necessary permissions to receive certificates from its respective PKI mount.
+  - controller and connect inject each locally update their mutatingwebhookconfigurations so that Kubernetes can relay events.
+  - Vault will manage certificate rotation and each of these webhooks will seamlessly received rotated certificates.
+
+To use Vault as the controller and connect inject Webhook Certificate Provider, we will need to modify the steps outlined in the [Data Integration](/docs/k8s/installation/vault/data-integration) section:
+
+### One time setup in Vault
+  1. Create a Vault policy that authorizes the desired level of access to the secret.
+
+### Setup per Consul datacenter
+  1. (Added) Create Vault PKI roles for controller and connect inject each that establish the domains that each is allowed to issue certificates for.
+  1. Create Vault Kubernetes auth roles that link the policy to each Consul on Kubernetes service account that requires access.
+  1. Configure the Vault Kubernetes auth roles in the Consul on Kubernetes helm chart.
+
+## One time setup in Vault
+
```

Comment:
```suggestion
You only need to perform the following steps once in Vault to ? 
```
I left a question mark because it's not clear what the one time setup is referring to. In any case, we need to introduce the following sections somehow.

Reply:
getting this data integrated means we integrate the systems.  Then we integrate the data.  When we integrate the data there are things that either need to be done once in vault or have to be done for every single data center you have.

So all of these vault pages try to paint the mental model since we are doing this in reference material and not a tutorial, to describe what you are doing and whether it is only once or is repeated for however many data centers you have.

This Vault stuff is very hard to write up in a reference style and this is the best that we have come up with so far to separate:
- system integration tasks vs data integration tasks
- one time set up vs repeat per data center

We are open to other ways to do this, but these vault docs are more involved in terms of a ton of manual steps than anything else probably in our kubernetes section.  So it has been difficult to get something that makes sense for everyone at first glance.

Reply:
I've moved this to the below section where it needs to be for each data center.